bin/check-all-gpg-keys-for-expiry.rb

```diff
@@ -0,0 +1,348 @@
+# description, level and number of days left
+class Key
+  attr_accessor :id, :expiry, :description, :level, :num_of_days_left
+
+  def initialize(id, expiry)
+    @id = id
+    @expiry = expiry
+  end
+
+  # Return the expiry date if @expiry is specified otherwise return
+  # 'N/A'
+  def display_expiry_date
+    @expiry.nil? ? 'N/A' : Time.at(@expiry.to_i).to_datetime.to_s
+  end
+
+  # Return the number of days left if @num_of_days_left is specified
+  # otherwise return 'N/A'
+  def display_num_of_days_left
+    @num_of_days_left == Integer::MAX ? 'N/A' : @num_of_days_left.to_s
+  end
+
+  # Return the meaningful information about the instance of Key
+  def inspect
+    'Id: ' + @id.to_s +
+      ', Expires on: ' + display_expiry_date +
+      ', Level: ' + map_number_to_word_level(@level) +
+      ', Number of days left to expire: ' + display_num_of_days_left +
+      ', Description: ' + @description % self
+  end
+
+  # Print the instance of Key's details in a nice format
+  def to_s
+    'Id: ' + @id.to_s +
+      ', Expires on: ' + display_expiry_date +
+      ', Level: ' + map_number_to_word_level(@level) +
+      ', Number of days left to expire: ' + display_num_of_days_left +
+      ', Description: ' + @description % self
+  end
+
+  private
+
+  # Return the meaningful word for a specified level
+  def map_number_to_word_level(level)
+    case level
+    when 0
+      'OK'
+    when 1
+      'WARNING'
+    when 2
+      'CRITICAL'
+    when 3
+      'UNKNOWN'
+    end
+  end
+end
+
+# == Description
+#
+# The CheckAllGpgKeysForExpiry class fetches all GPG keys, check their expiry dates and send
+# appropriate alerts to Sensu
+class CheckAllGpgKeysForExpiry < Sensu::Plugin::Check::CLI
+  option :warning,
+         short: '-w NUMBER_OF_DAYS_LEFT_TO_TRIGGER_ALERT',
+         long: '--warning NUMBER_OF_DAYS_LEFT_TO_TRIGGER_ALERT',
+         proc: proc(&:to_i),
+         default: 14,
+         description: 'The minimum number of days left to trigger a warning alert'
+
+  option :critical,
+         short: '-c NUMBER_OF_DAYS_LEFT_TO_TRIGGER_ALERT',
+         long: '--critical NUMBER_OF_DAYS_LEFT_TO_TRIGGER_ALERT',
+         proc: proc(&:to_i),
+         default: 5,
+         description: 'Minimum number of days left to trigger a critical alert'
+
+  option :home_directory,
+         short: '-h GPG_HOME_DIR',
+         long: '--home-directory GPG_HOME_DIR',
+         default: '/root/.gnupg',
+         description: 'Location of GPG home directory'
+
+  option :gpg,
+         short: '-e GPG_EXECUTABLE_PATH',
+         long: '--execute GPG_EXECUTABLE_PATH',
+         default: '/usr/bin/gpg',
+         description: 'Location of binary executable file (gpg)'
+
+  option :use_secret_key,
+         short: '-s (yes|no)',
+         long: '--use-secret-key (yes|no)',
+         default: 'yes',
+         description: 'Check secret key instead of public key'
+
+  option :do_not_display_json_message,
```

Comment:
Is there a reason we don't just make this a boolean?

Reply:
I was using old version of mixlib-cli which does not support boolean. I have updated `:do_not_display_json_message` to use boolean.